api/v1alpha1/clienttrafficpolicy_types.go

```diff
@@ -89,11 +89,20 @@ type ClientTrafficPolicySpec struct {
 }
 
 // ClientIPDetectionSettings provides configuration for determining the original client IP address for requests.
+//
+// +kubebuilder:validation:XValidation:rule="!(has(self.xForwardedFor) && has(self.customHeader))",message="customHeader cannot be used in conjunction with xForwardedFor"
```

Comment:
That is correct. The top-level `xff_num_trusted_hops` knob in Envoy is mutually exclusive with `original_ip_detection_extensions`. Even if Envoy supported both simultaneously though, I'm not sure it would make sense to have two methods of downstream client IP detection configured at the same time as it would muddy where the logged downstream IP address came from.

When `customHeader` is specified, any XFF headers are essentially ignored.

Reply:
> One of 'xForwordFor' and 'customHeader' must be specified?

Can we also add a CEL validation for that?

Reply:
@zhaohuabing I think I misunderstood your question at first. They are mutually exclusive, but neither one is required. When both are omitted, they are simply not configured and the default behavior is to use the remote address which is correct when Envoy is directly exposed to clients rather than residing behind a CDN or cloud-provider LB. Not sure any additional CEL validation is needed here.